network/retrieval/retrieve.go

```diff
@@ -187,6 +189,111 @@ func (r *Retrieval) getOriginPo(req *storage.Request) int {
 	return originPo
 }
 
+// findPeer finds a peer we need to ask for a specific chunk from according to our kademlia
+func (r *Retrieval) findPeerLB(ctx context.Context, req *storage.Request) (retPeer *network.Peer, err error) {
```

Comment:
This function shares (copies) code with findPeer function. Could code duplication be reduced?

Reply:
this is intentional since i did not want to touch the original logic and i did not want to change the structure of the function. i stated this in the pr description. obviously some of it can be deduplicated but it means changing both of these functions and i wanted to have a commit that can be easily reverted if this change causes any trouble down the line

Reply:
If that is the argument for code duplication, then testing should be improved to ensure that the logic in function is correct in respect to the expected behaviour.

I do not see any point keeping `findPeer` function as it is not used anymore. It would be even easier to review changes with diff to the same function, then comparing one function side by side to another as you suggested.

I am in favour of merging better designed code instead copies of similar code. Having a better designed code is more important than protecting ourselves from troubles with reverting code.